hstrat/phylogenetic_inference/tree/_impl/_TrieInnerNode.py

```diff
@@ -241,6 +241,26 @@ def taxon_label(self: "TrieInnerNode") -> str:
                 render_to_base64url(int(self._tiebreaker))
             }"""
 
+    # recursive check
+    def __eq__(self: "TrieInnerNode", other: object) -> bool:
```

Comment:
add a short docstring explaining how equality is determined (i.e., elaborate on "recursive check")